src/openpgp.js

```diff
@@ -63,10 +64,12 @@ export async function generateKey({ userIDs = [], passphrase = '', type = 'ecc',
   if (type === 'rsa' && rsaBits < config.minRSABits) {
     throw new Error(`rsaBits should be at least ${config.minRSABits}, got: ${rsaBits}`);
   }
+
   const options = { userIDs, passphrase, type, rsaBits, curve, keyExpirationTime, date, subkeys };
 
   try {
     const { key, revocationCertificate } = await generate(options, config);
+    checkKeyRequirements(key.keyPacket, config);
```

Comment:
This is the more elegant solution, since before generation we don't necessarily have the final algo + curve pairings. The downside of checking the key after generation, is that the thrown error will differ depending on the platform and config options. e.g. if someone disables `config.useIndutnyElliptic`, and leaves the brainpool curves blacklisted, the error will be `This curve is only supported in the full build of OpenPGP.js` (from `generate`), rather than `... support for curve XXX is disabled`.

Reply:
I'm not opposed to this, but probably we should check all subkey packets as well here?